client/script/test.bash

```diff
@@ -6,9 +6,20 @@ PATH="$(yarn bin):$PATH"
 export PATH
 jest .
 
-# { { yarn start 2>&3 || [ $? -eq 137 ]; } 3>&2 2>/dev/null & } <-- Does not work. Trying to only suppress error 137.
-yarn start >/dev/null & # Start server in background. Suppress stderr.
-printf "\n\n################ Running Playwright ################"
-playwright test
-printf "Closing server on port 4000..."
-npx kill-port 4000
+if [ -z "${1+x}" ]; then
```

Comment:
what does this line do? A switch case across `u, i , e, a` is probably simpler.

Reply:
The line makes it so if the user dosent pass a argument (-a or -e) it will run the default mode which is unit tests, however if a argument is passed and its "-a" or "-e" it will also run the e2e tests. Would you like me to change this so a argument is required? Also we do not have any integration tests so would "-i" be required?

Reply:
A switch statement will have a default case. You can put the unit tests in the default case. If you don't use the break statement in switch, the default will always run. That should take care of your intent.

It seems don't have the integration tests but they are a good protection against unintended code changes. Is there a way to test the following integrations?
1. Redux and authentication code synchronization of username before login, after login and after logout

We talked about integration tests with gitlab. Are they in the PR now? If not, please include them under `-eg`, i.e., e2e with gitlab. You can skip this test on github actions.

Reply:
Okay yes I see that a switch statement would be ideal here, I will update this.

A integration test between Redux and authentication sounds like a good idea, I will look into this but I might have to delay adding it if it takes extra time because of the bachelor deadline.

Currently the e2e tests are still in e2e folder but I can certainly make a e2e-gitlab folder, I am just not sure what to add here as we can't make any of the e2e tests work (Nock did not work as discussed previously). I can look into making a specific e2e test with GitLab but it might take some time.

Reply:
ok, let's skip the the integration tests.

Reply:
Hi Prasad just to clarify, would you like me to skip all the integration tests, the integration tests with GitLab (e2e with GitLab) or the integration test of Redux and authentication code synchronization of username?

I think the integration test of Redux and authentication code synchronization of username is a good idea but the integration tests with GitLab (e2e with GitLab) seems rather complicated.